index.bs

```diff
@@ -353,330 +424,229 @@ For the main <dfn>sanitize</dfn> operation, using a {{ParentNode}} |node|, a
+1. [=SanitizerConfig/Remove=] |element| from |configuration|["{{SanitizerConfig/removeElements}}"].
+1. [=SanitizerConfig/Remove=] |element| from |configuration|["{{SanitizerConfig/elements}}"] list.
+
+</div>
+
+<div algorithm>
+To <dfn for="Sanitizer">allow an attribute</dfn> |attribute| on a {{SanitizerConfig}} |configuration|, do:
+
+1. Set |attribute| to the result of [=canonicalize a sanitizer attribute=] with |attribute|.
+1. [=SanitizerConfig/Add=] |attribute| to |configuration|["{{SanitizerConfig/attributes}}"].
+1. [=SanitizerConfig/Remove=] |attribute| from |configuration|["{{SanitizerConfig/removeAttributes}}"].
+
+</div>
+
+<div algorithm>
+To <dfn for="Sanitizer">remove an attribute</dfn> |attribute| from a {{SanitizerConfig}} |configuration|, do:
+
+1. Set |attribute| to the result of [=canonicalize a sanitizer attribute=] with |attribute|.
+1. [=SanitizerConfig/Add=] |attribute| to |configuration|["{{SanitizerConfig/removeAttributes}}"].
+1. [=SanitizerConfig/Remove=] |attribute| from |configuration|["{{SanitizerConfig/attributes}}"].
+
+</div>
+
+<div algorithm>
+To <dfn for="Sanitizer">set comments</dfn> with |allow| on a {{SanitizerConfig}} |configuration|, do:
+
+1. Set |configuration|["{{SanitizerConfig/comments}}"] to |allow|.
+
+</div>
+
+<div algorithm>
+To <dfn for="Sanitizer">set data attributes</dfn> with |allow| on a {{SanitizerConfig}} |configuration|, do:
+
+1. Set |configuration|["{{SanitizerConfig/dataAttributes}}"] to |allow|.
+
+</div>
+
+<div algorithm>
+
+Note: While this algorithm is called [=remove unsafe=], we use
+    <a href="#security-considerations">the term "unsafe" strictly in the sense
+    of this spec</a>, to denote content that will
+    execute JavaScript when inserted into the document. In other words, this
+    method will remove oportunities for XSS.
+
+To <dfn for="SanitizerConfig">remove unsafe</dfn> from a |configuration|, do this:
+
+1. [=Assert=]: The [=built-in safe baseline config=] has
+   {{SanitizerConfig/removeElements}} and {{SanitizerConfig/removeAttributes}}
+   keys set, but not {{SanitizerConfig/elements}},
+   {{SanitizerConfig/replaceWithChildrenElements}}, or
+   {{SanitizerConfig/attributes}}.
+1. Let |result| be a copy of |configuration|.
+1. [=list/For each=] |element| in
+   [=built-in safe baseline config=][{{SanitizerConfig/removeElements}}]:
+    1. Call [=remove an element=] with |element| and |result|.
+1. [=list/For each=] |attribute| in
+   [=built-in safe baseline config=][{{SanitizerConfig/removeAttributes}}]:
+    1. Call [=Sanitizer/remove an attribute=] with |attribute| and |result|.
 1. Return |result|.
 
 </div>
 
 <div algorithm>
-In order to <dfn>canonicalize a sanitizer element list</dfn> |list|, with a
-default namespace |defaultNamespace|, run the following steps:
+To <dfn for="Sanitizer">set a config</dfn> |configuration| on a {{Sanitizer}} |sanitizer|, do this:
+
+1. [=Assert=]: |configuration| is a [=dictionary=].
```

Comment:
Instead of asserting this here, you should specify the argument type above.

> To **set a config** given a dictionary |configuration| and a Sanitizer |sanitizer|:

Reply:
Done. That indeed reads nicer.